File: moderate/largest_sum.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    my @s = split( /,/, $line );
    my ( $l, $m ) = ( 0, @s[0] );
    for (@s) {
        $m = $_ + $l if $_ + $l > $m;
        $l = $_ + $l > 0 ? $_ + $l : 0;
    }
    print "$m\n";
}
close(INFILE);

File: moderate/largest_sum.rb
File.open(ARGV[0]).each_line do |line|
  s = line.split(',').map(&:to_i)
  l = 0
  m = s[0]
  s.each do |i|
    m = i + l if i + l > m
    l = i + l > 0 ? i + l : 0
  end
  puts m
end
